File: backend/Payments/src/Payments/Controllers/TransactionsController.cs
using Microsoft.AspNetCore.Mvc;
using Payments.Application.Common.Models;
using Payments.Application.Transactions.Commands.CancelScheduledBillCommand;
using Payments.Application.Transactions.Commands.PayBillNowCommand;
using Payments.Application.Transactions.Commands.PayBillOnSpecificDateCommand;
using Payments.Application.Transactions.Dto;
using Payments.Application.Transactions.Queries.GetTransactionsWithPagination;

namespace Payments.Controllers;

public class TransactionsController: ApiControllerBase
{
    [HttpGet]
    public async Task<PaginatedList<TransactionModel>> GetTransactionsAsync([FromQuery]GetTransactionsWithPaginationQuery query)
    {
        return await Mediator.Send(query);
    }
    
    [HttpPost("paynow")]
    public async Task<IActionResult> PayBillNow(PayBillDto payBillDto)
    {
        return Ok(await Mediator.Send(new PayBillNowCommand(payBillDto)));
    }

    [HttpPost("payondate")]
    public async Task<IActionResult> PayBillOnDate(PayBillDto payBillDto)
    {
        return Ok(await Mediator.Send(new PayBillOnSpecificDateCommand() { PayBillDto = payBillDto } ));
    }

    [HttpPost("cancel")]
    public async Task<IActionResult> CancelBill(CancelBillDto model)
    {
        return Ok(await Mediator.Send(new CancelScheduledBillCommand () { CancelBillDto = model }));
    }
}

File: backend/PaymentsNest/src/modules/transactions/transactions.controller.ts
import { Body, Controller, Get, Post, Query } from '@nestjs/common';
import { GetTransactionsQuery } from './models/get-transactions.query';
import { TransactionsService } from './transactions.service';
import { PayBillDTO } from './models/pay-bill.model';

@Controller('Transactions')
export class TransactionsController {
  constructor(private readonly transactionsService: TransactionsService) {}

  @Get()
  async getTransactions(@Query() query: GetTransactionsQuery) {
    return await this.transactionsService.getTransactions(query);
  }

  @Post('paynow')
  async payNow(@Body() payBillDTO: PayBillDTO) {
    return await this.transactionsService.payNow(payBillDTO);
  }
}
